src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java

```diff
@@ -0,0 +1,134 @@
+    }
+
+    @Nullable
+    @Override
+    public Path which(String fn) {
+        String ext = Utils.extension(fn);
+        if (isAbsolutePath(fn)) {
+            if (!ext.isEmpty()) {
+                Path f = Paths.get(fn);
+                return Files.isExecutable(f) ? f : null;
+            }
+            // No extension provided. Try PATHEXT in order
+            for (String extCandidate : PATHEXT) {
+                Path f = Paths.get(fn + extCandidate);
+                if (Files.isExecutable(f)) {
+                    return f;
+                }
+            }
+            return null;
+        }
+        // Search in paths
+        for (Path d : paths) {
+            Path f = d.resolve(fn);
+            if (!ext.isEmpty() && Files.isExecutable(f)) {
+                return f;
+            }
+            // No extension provided. Try PATHEXT in order
+            for (String extCandidate : PATHEXT) {
+                f = d.resolve(fn + extCandidate);
+                if (Files.isExecutable(f)) {
+                    return f;
+                }
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public String[] getCommand() {
+        String[] decorated = cmds;
+        if (shellDecorator != null) {
+            decorated = shellDecorator.decorate(decorated);
+        }
+        // First item in the command is the executable. If it's given as absolute path, add quotes around it
+        // in case the path contains space which will break the whole command line
+        // See security remarks:
+        // https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-createprocesswithlogonw
+        if (isAbsolutePath(decorated[0])) {
+            decorated[0] = String.format("\"%s\"", decorated[0]);
+        }
+        return decorated;
+    }
+
+    @Override
+    protected Process createProcess() {
+        WindowsRunasProcess winProcess = new WindowsRunasProcess(null, userDecorator.getUser());
+        winProcess.setLpEnvironment(computeEnvironmentBlock());
+        winProcess.setLpCurrentDirectory(dir.getAbsolutePath());
+        winProcess.start(String.join(" ", getCommand()));
+        return winProcess;
+    }
+
+    @Override
+    public synchronized void close() throws IOException {
+        if (isClosed.get()) {
+            return;
+        }
+        if (process == null || !process.isAlive()) {
+            return;
+        }
+        // TODO first try to shutdown process and children gracefully
+        // Then force kill if not stopped within timeout
+        Platform platformInstance = Platform.getInstance();
+        try {
+            platformInstance.killProcessAndChildren(process, false, Collections.emptySet(), userDecorator);
+        } catch (InterruptedException e) {
+            // If we're interrupted make sure to kill the process before returning
+            try {
+                platformInstance.killProcessAndChildren(process, true, Collections.emptySet(), userDecorator);
```

Comment:
put this inside a `finally` instead of the `catch` that way it always runs 100% of the time.